src/evict/evict_lru.c

```diff
@@ -27,24 +27,31 @@ static int  __evict_server_work(WT_SESSION_IMPL *);
 static inline uint64_t
 __evict_read_gen(const WT_EVICT_ENTRY *entry)
 {
+	WT_BTREE *btree;
 	WT_PAGE *page;
 	uint64_t read_gen;
 
+	btree = entry->btree;
+
 	/* Never prioritize empty slots. */
 	if (entry->ref == NULL)
 		return (UINT64_MAX);
 
 	page = entry->ref->page;
 
+	/* Any page from a dead tree is a great choice. */
+	if (F_ISSET(btree->dhandle, WT_DHANDLE_DEAD))
+		return (WT_READGEN_OLDEST);
```

Comment:
I don't know if I'm that enthusiastic. How about limiting this to leaf pages? It's wasted effort if there are live children.